sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -422,7 +422,7 @@ object TypeCoercion {
 
         val commonTypes = lhs.zip(rhs).flatMap { case (l, r) =>
           findCommonTypeForBinaryComparison(l.dataType, r.dataType)
-            .orElse(findTightestCommonType(l.dataType, r.dataType))
+            .orElse(findWiderTypeForTwo(l.dataType, r.dataType))
```

Comment:
Why we make this change?

Reply:
To be coherent with what is done when there are literals instead of a subquery.